src/game/boe.items.cpp

```diff
@@ -313,17 +313,24 @@ void set_town_attitude(short lo,short hi,eAttitude att) {
 	univ.town.monst.hostile = true;
 	long long num_monst = univ.town.monst.size();
 	
-	// Nice smart indexing, like Python :D
-	if(lo <= -num_monst)
-		lo = 0;
-	if(lo < 0)
-		lo = num_monst + lo;
-	if(hi <= -num_monst)
-		hi = 0;
-	if(hi < 0)
-		hi = num_monst + hi;
-	if(hi < lo)
-		std::swap(lo, hi);
+	if (lo==-1 && hi==-1) {
```

Comment:
Why does this mean all monsters? Is there something that uses this?

Reply:
I probably tried to correct a problem with the old special 170 which normally called :
```
		case 170:
			make_town_hostile();
			break;
```
currently we have
```
		case 170: type = eSpecType::MAKE_TOWN_HOSTILE; break;
			// 171-173 were change/swap/transform terrain
```
and
```
		case eSpecType::MAKE_TOWN_HOSTILE:
			if(spec.ex2a < 0 || spec.ex2a > 3){
				showError("Invalid attitude (0-Friendly Docile, 1-Hostile A, 2-Friendly Will Fight, 3-Hostile B).");
				break;
			}
			set_town_attitude(spec.ex1a,spec.ex1b,eAttitude(spec.ex2a));
			break;
```

So if the code does no change the values of ex1a and ex1b, we call set_town_attitude(-1,-1, eAttitude(spec.ex2a)); which kills almost nobody. But yes, it would be better to change:
```
		case 170: type = eSpecType::MAKE_TOWN_HOSTILE; ex1a=0; ex1b=-1; break;
```

Reply:
Okay, I went ahead and did that.